src/libmongoc/tests/test-mongoc-client-side-encryption.c

```diff
@@ -2321,16 +2324,17 @@ test_bypass_spawning_via_mongocryptdBypassSpawn (void *unused)
 
    /* Create a MongoClient with encryption enabled */
    client_encrypted = test_framework_new_default_client ();
-   extra =
-      BCON_NEW ("mongocryptdBypassSpawn",
-                BCON_BOOL (true),
-                "mongocryptdSpawnArgs",
-                "[",
-                "--pidfilepath=bypass-spawning-mongocryptd.pid",
-                "--port=27021",
-                "]",
-                "mongocryptdURI",
-                "mongodb://localhost:27021/?serverSelectionTimeoutMS=1000");
+   extra = BCON_NEW ("mongocryptdBypassSpawn",
+                     BCON_BOOL (true),
+                     "mongocryptdSpawnArgs",
+                     "[",
+                     "--pidfilepath=bypass-spawning-mongocryptd.pid",
+                     "--port=27021",
+                     "]",
+                     "mongocryptdURI",
+                     "mongodb://localhost:27021/?serverSelectionTimeoutMS=1000",
+                     "__csfleDisabled",
```

Comment:
All drivers will need a solution to skip this prose test when running with `csfle`. I prefer not adding a hidden option to the API. I think it adds small risk that users rely on it.

I suggest only run this test if the test runner expects to use `csfle`. That could be passed through an environment variable. There is precedence: (e.g. `MONGOC_TEST_IS_SERVERLESS`, `MONGOC_TEST_LOADBALANCED`.